hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataFileSystemView.java

```diff
@@ -73,4 +73,16 @@ public void close() {
       throw new HoodieException("Error closing metadata file system view.", e);
     }
   }
+
+  @Override
+  public void sync() {
+    // Sync the tableMetadata first as super.sync() may call listPartition
+    if (tableMetadata != null) {
+      BaseTableMetadata baseMetadata = (BaseTableMetadata)tableMetadata;
```

Comment:
nit: space after `)`. Suprised that checkstyle is happy

Reply:
Done.